bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/handler/RoomHandler.java

```diff
@@ -112,30 +131,58 @@ public void refresh(@Nullable QueryResponseDTO domain) {
         }
     }
 
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(WarmupActions.class);
+    }
+
     private void setOverride(final QuantityType<?> command) {
-        String roomId = getThing().getProperties().get(PROPERTY_ROOM_ID);
-        String locationId = getThing().getProperties().get(PROPERTY_LOCATION_ID);
+        setOverride(command, new QuantityType<>(config.getOverrideDuration(), Units.MINUTE));
+    }
 
-        QuantityType<?> temp = command.toUnit(SIUnits.CELSIUS);
+    public void setOverride(final QuantityType<?> temperature, final QuantityType<?> duration) {
+        setOverride(formatTemperature(temperature), duration.toUnit(Units.MINUTE).intValue());
+    }
 
-        if (temp != null) {
-            final int value = temp.multiply(BigDecimal.TEN).intValue();
+    private void setOverride(final int temperature, final int duration) {
+        String roomId = getThing().getProperties().get(PROPERTY_ROOM_ID);
```

Comment:
`getThing().getProperties()` could be factored out : `properties = getThing().getProperties();`